### PR TITLE
Change ?amp to ?amp=1 

### LINKED ISSUE #1383: Use query arg 'amp=1' instead of 'amp' for caching purposes
Usually the static cache configuration (nginx, static html files, proxy, etc) tells the server to bypass cache if the query string is not empty. Since 1.0 the amp URLs use ?amp and that prevents the server from caching it.

It's really easy to add a rule that tells the server to cache a page if  a non-empty query arg is present but it's not easy to work with empty query args.

If we change ?amp to ?amp=1 it will have the same effect in rewrite rules and will make it possible to the server detect if amp query arg is present and deliver cached content.

### PR DESCRIPTION
**Pull request by ricardobrg**
Usually the static cache configuration (nginx, static html files, proxy, etc) tells the server to bypass cache if the query string is not empty. Since 1.0 the amp URLs use ?amp and that prevents the server from caching it.

It's really easy to add a rule that tells the server to cache a page if a non-empty query arg is present but it's not easy to work with empty query args.

I've changed `amp` query var to `amp=1` to make it possible to the server or cache and CDN services detect if amp query arg is present and deliver cached content.

Fixes #1383.

### CODE REVIEW

File: includes/amp-helper-functions.php [hunk 1]
```diff
@@ -85,7 +85,7 @@ function amp_get_permalink( $post_id ) {
 	if ( current_theme_supports( 'amp' ) ) {
 		$permalink = get_permalink( $post_id );
 		if ( ! amp_is_canonical() ) {
-			$permalink = add_query_arg( 'amp', '', $permalink );
+			$permalink = add_query_arg( 'amp', '1', $permalink );
```
Comment (westonruter) [inline includes/amp-helper-functions.php]: If we go ahead with this PR, then what we should do is take the opportunity to factor out the redundant formation of the AMP URLs with a new common function. There is presently `amp_get_permalink()` but that is specifically for getting URLs for posts. The `amp_get_url()` function would be used inside of `amp_get_permalink()`.

There could be an `amp_get_url()` function that looks like this:

```php
/**
 * Get the AMP version of a URL.
 *
 * This simply appends the required query parameter to the URL. It does not confirm
 * whether or not AMP is allowed on the given URL. If AMP is not supported, then a redirect
 * to the non-AMP version will ensue.
 *
 * @see amp_get_permalink()
 * @return string AMP URL.
 */
function amp_get_url( $url ) {
    return add_query_arg( amp_get_slug(), '1', $url );
}
```

Comment (ricardobrg) [reply includes/amp-helper-functions.php]: @westonruter  I'll work on it...

Comment (ricardobrg) [reply includes/amp-helper-functions.php]: Hello @westonruter! Getting back to this now...

**Comment by swissspidy** [comment]
@ricardobrg Would you still be interested in bringing this into the finish line? 🙂

**Comment by ricardobrg** [comment]
@swissspidy yes! It is in my "todo list". I need to update the code and work in the feature. Give me one more week and I'll have another commit to this PR.

**Comment by westonruter** [comment]
I think we'll need to step back and re-consider this at a higher level. There are some negative side effects to changing the AMP URLs. See #2204. I suggest we close this PR and take into account all the points in that issue in a new PR. It will require deeper changes than just changing the URLs.

**Comment by ricardobrg** [comment]
Since this PR is almost 1 year old and lots of changes were made in the plugin, it will need a deeper work anyway. I will follow @westonruter suggestion and start working from #2204 considering the other related issues.

**Pull request closed without merging**